kornia/losses/focal.py

```diff
@@ -83,7 +83,7 @@ def forward(  # type: ignore
 
         # compute the actual focal loss
         weight = torch.pow(torch.tensor(1.) - input_soft,
-                           self.gamma.to(input.dtype))
+                           self.gamma.to(input.device))
```

Comment:
@leaxp thanks for the contribution. I believe we might need both  to type and device, otherwise the gradcheck tests won't pass. We are working on integrating gpu tests, so expected unexpected behaviors on that end.

Reply:
We definitely need both